projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -246,13 +246,22 @@ export class TuiInputPhoneInternationalComponent
                 /^(?!880[1-9 ])/.test(value) &&
                 value.length + 1 === this.getMaxAllowedLength(TuiCountryIsoCode.RU);
 
-            return (
+            const matched =
                 ruCodeTest ||
                 (value.startsWith(
                     this.isoToCountryCode(countryIsoCode).replace(CHAR_PLUS, ''),
                 ) &&
-                    value.length + 1 === this.getMaxAllowedLength(countryIsoCode))
-            );
+                    value.length + 1 === this.getMaxAllowedLength(countryIsoCode));
+
+            return matched && countryIsoCode === TuiCountryIsoCode.RU
+                ? this.ruCodeShouldNotBeIncludeKzRegion(value)
```

Comment:
1. Let's not make long names (`notKzRegion` should be ok)
2. Let's move it to a function so the name is minified.

Reply:
fixed